object-store/src/main/java/org/hypertrace/config/objectstore/DefaultObjectStore.java

```diff
@@ -19,14 +21,17 @@
   private final ConfigServiceBlockingStub configServiceBlockingStub;
   private final String resourceNamespace;
   private final String resourceName;
+  private final ConfigChangeEventGenerator configChangeEventGenerator;
 
   protected DefaultObjectStore(
       ConfigServiceBlockingStub configServiceBlockingStub,
       String resourceNamespace,
-      String resourceName) {
+      String resourceName,
+      ConfigChangeEventGenerator configChangeEventGenerator) {
```

Comment:
Could we overload this to make it optional? That way, implementers can opt out (and we save ourselves from this being a breaking change)